src/Debug/skeletonViewer.ts

```diff
@@ -666,19 +667,25 @@ export class SkeletonViewer {
                         updatable: false
                     },  scene);
 
-                    spur.convertToFlatShadedMesh();
-
                     let numVertices = spur.getTotalVertices();
                     let mwk: number[] = [], mik: number[] = [];
 
                     for (let i = 0; i < numVertices; i++) {
                         mwk.push(1, 0, 0, 0);
-                        mik.push(bone.getIndex(), 0, 0, 0);
+
+                        if (displayOptions.spurFollowsChild && i > 9) {

```

Comment:
I think `i > 9` was for debugging purpose?

Reply:
No, that's the logic that selects the vertices to be bound to the child bone.

Reply:
The spur consists of an extruded square of five vertices. It is extruded in three steps. The first square (vert 0 to 4) are on the bone, the second square (vert 5 to 9) are on the `midStep`. The last square end up on the child bone (vert 10 to 14).

This how I interpret the code.

I moved down the call to `convertToFlatShadedMesh` since it modifies the mesh and makes it almost impossible to select the vertices that end up on the child bone. This is also a tiny optimization I think, since less weights are needed if we add them before converting to flat shading. `convertToFlatShadedMesh` then automatically updates all vertex indices.

Reply:
Perhaps a code comment is warranted.

Reply:
Yes just a code comment to avoid future questions :D